app/services/risc_destination_updater.rb

```diff
@@ -24,27 +24,36 @@ def remove
     eventbridge_client.delete_connection(name: connection_name) if connection_exists?
   end
 
+  def issuer_slug
+    service_provider.issuer.gsub(/[^.\-_A-Za-z0-9]/, '_')
+  end
+
   def connection_name
-    "#{Identity::Hostdata.env}-risc-connection-#{service_provider.issuer}"
+    "#{Identity::Hostdata.env}-risc-connection-#{issuer_slug}"
   end
 
   def rule_name
-    "#{Identity::Hostdata.env}-risc-rule-#{service_provider.issuer}"
+    "#{Identity::Hostdata.env}-risc-rule-#{issuer_slug}"
   end
 
   def api_destination_name
-    "#{Identity::Hostdata.env}-risc-destination-#{service_provider.issuer}"
+    "#{Identity::Hostdata.env}-risc-destination-#{issuer_slug}"
   end
 
   def target_id
-    "#{Identity::Hostdata.env}-risc-target-#{service_provider.issuer}"
+    "#{Identity::Hostdata.env}-risc-target-#{issuer_slug}"
   end
 
   def event_bus_name
     # Matches value managed by Terraform
     "#{Identity::Hostdata.env}-risc-notifications"
   end
 
+  def destination_iam_role_arn
+    # Matches value managed by Terraform
+    "arn:aws:iam::#{aws_account_id}:role/#{Identity::Hostdata.env}-risc-notification-destination"
```

Comment:
Any way we could incorporate a test assertion here? Partly also wondering if `::` is expected?

Reply:
so it turns out... the `::` is not expected it was a copy-paste error on my part, but yes will add a test case

Reply:
I got some ARNs confused... I believe this *should* have the `::`... added a spec in 89763b0fa39693d2dafb7fd76f1a897fa2e733f8, [slack message why i think this is right](https://gsa-tts.slack.com/archives/C01D7MWNE64/p1624397530017100?thread_ts=1624374211.009400&cid=C01D7MWNE64)